From 50c6bdbea3f3c28c3c6b101e2ca0f7e3c676e4a2 Mon Sep 17 00:00:00 2001
From: Jian-Hong Pan <chienhung.pan@gmail.com>
Date: Sat, 14 Oct 2023 16:27:13 +0800
Subject: [PATCH] [GTK] Bump WebKit2 to 4.1

According to "WebKitGTK API for GTK 4 Is Now Stable" [1], webkit2gtk-4.0
is obsolete and users should immediately port to webkit2gtk-4.1.
Besides, webkit2gtk-4.1 using GTK 3 and libsoup 3 has become the default
API version in WebKitGTK 2.40 [2].

The GNOME 44(+) flatpak runtimes have already dropped webkit2gtk-4.0:

$ flatpak run --command=find org.gnome.Platform//44 /lib/x86_64-linux-gnu/ -name libwebkit*
/lib/x86_64-linux-gnu/webkitgtk-6.0/injected-bundle/libwebkitgtkinjectedbundle.so
/lib/x86_64-linux-gnu/libwebkitgtk-6.0.so.4
/lib/x86_64-linux-gnu/libwebkit2gtk-4.1.so.0.12.4
/lib/x86_64-linux-gnu/libwebkit2gtk-4.1.so.0
/lib/x86_64-linux-gnu/libwebkitgtk-6.0.so.4.4.4
/lib/x86_64-linux-gnu/webkit2gtk-4.1/injected-bundle/libwebkit2gtkinjectedbundle.so

PS. GNOME 43 flatpak runtime has been end-of-life.

And, most Linux distributions support WebKit2 4.1 now.
* Arch Linux: https://archlinux.org/packages/?sort=&q=webkit2gtk&maintainer=&flagged=
* Alpine: https://pkgs.alpinelinux.org/packages?name=webkit2gtk*&branch=v3.18&repo=&arch=&maintainer=
* Debian: https://packages.debian.org/search?keywords=webkit2gtk
* Ubuntu: https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=webkit2gtk&searchon=names
* Fedora: https://packages.fedoraproject.org/pkgs/webkitgtk/
* ...

[1]: https://blogs.gnome.org/mcatanzaro/2023/03/21/webkitgtk-api-for-gtk-4-is-now-stable/
[2]: https://github.com/WebKit/WebKit/blob/wpewebkit-2.40.0/Source/cmake/OptionsGTK.cmake#L232-L235

PR: https://github.com/r0x0r/pywebview/pull/1234
---
 webview/platforms/gtk.py   | 4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/webview/platforms/gtk.py b/webview/platforms/gtk.py
index 27a027c..5beb0f7 100755
--- a/webview/platforms/gtk.py
+++ b/webview/platforms/gtk.py
@@ -21,8 +21,8 @@ import gi
 
 gi.require_version('Gtk', '3.0')
 gi.require_version('Gdk', '3.0')
-gi.require_version('WebKit2', '4.0')
-gi.require_version('Soup', '2.4')
+gi.require_version('WebKit2', '4.1')
+gi.require_version('Soup', '3.0')
 
 from gi.repository import Gdk, Gio
 from gi.repository import GLib as glib
-- 
2.42.0

